src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -651,12 +651,24 @@ where
         request: Request<DropConnectionRequest>,
     ) -> Result<Response<DropConnectionResponse>, Status> {
         let req = request.into_inner();
+        let connection = self
+            .catalog_manager
+            .get_connection_by_id(req.connection_id)
+            .await?;
 
         let version = self
             .ddl_controller
-            .run_command(DdlCommand::DropConnection(req.connection_id))
+            .run_command(DdlCommand::DropConnection(connection.id))
             .await?;
 
+        // delete AWS vpc endpoint
+        if let Some(connection::Info::PrivateLinkService(svc)) = connection.info
+            && svc.get_provider()? == PbPrivateLinkProvider::Aws {
+            if let Some(aws_cli) = self.aws_client.as_ref() {
+                aws_cli.delete_vpc_endpoint(&svc.endpoint_id).await?;
```

Comment:
If deleting vpc endpoint fails here, the frontend will receive an error but the connection is already dropped, which will make the ddl operation unretriable. Shall we just print a warning here and leave it to be handled somewhere else maybe in cloud side. 🤔

Reply:
It also could happen when meta crashes before deleting the endpoint.

Reply:
I also notice that we don't provide atomic semantic to users here. If Meta crashed or the delete vpc operation failed, the vpc endpoint will be leaked and need to be deleted manually. So it is a best-effort semantic. Is it acceptable to you? @mikechesterwang @wjf3121

Reply:
I found that @mikechesterwang have a suggestion in the slack channel that is to add a `tags` field for connection, which will be used for periodic resource leak check in cloud side. That mechanism LGTM. We can implement it in the next PR.